### ISSUE TITLE
DiffExTest.testCountData and MeanVarianceServiceTest.testServiceCreateCountData often fail due to org.springframework.security.access.AccessDeniedException: Access is denied

### ISSUE THREAD

**Issue by arteymix** [PavlidisLab/Gemma#206]
It might be caused by missing ACLs entries since I can see these errors from the CI:

```
[gemma-core 2021-10-12 22:03:50,700] WARN [main] gemma.gsec.SecurityServiceImpl.hasPermission(667) | Unable to find ACL information for object identity 'gemma.gsec.acl.domain.AclObjectIdentity[Type: ubic.gemma.model.expression.bioAssay.BioAssay; Identifier: 1438]'
[gemma-core 2021-10-12 22:03:50,708] WARN [main] gemma.gsec.SecurityServiceImpl.hasPermission(667) | Unable to find ACL information for object identity 'gemma.gsec.acl.domain.AclObjectIdentity[Type: ubic.gemma.model.expression.bioAssay.BioAssay; Identifier: 1439]'
[gemma-core 2021-10-12 22:03:50,715] WARN [main] gemma.gsec.SecurityServiceImpl.hasPermission(667) | Unable to find ACL information for object identity 'gemma.gsec.acl.domain.AclObjectIdentity[Type: ubic.gemma.model.expression.bioAssay.BioAssay; Identifier: 1440]'
[gemma-core 2021-10-12 22:03:50,720] WARN [main] gemma.gsec.SecurityServiceImpl.hasPermission(667) | Unable to find ACL information for object identity 'gemma.gsec.acl.domain.AclObjectIdentity[Type: ubic.gemma.model.expression.bioAssay.BioAssay; Identifier: 1441]'
[gemma-core 2021-10-12 22:03:50,725] WARN [main] gemma.gsec.SecurityServiceImpl.hasPermission(667) | Unable to find ACL information for object identity 'gemma.gsec.acl.domain.AclObjectIdentity[Type: ubic.gemma.model.expression.bioAssay.BioAssay; Identifier: 1434]'
[gemma-core 2021-10-12 22:03:50,731] WARN [main] gemma.gsec.SecurityServiceImpl.hasPermission(667) | Unable to find ACL information for object identity 'gemma.gsec.acl.domain.AclObjectIdentity[Type: ubic.gemma.model.expression.bioAssay.BioAssay; Identifier: 1436]'
[gemma-core 2021-10-12 22:03:50,739] WARN [main] gemma.gsec.SecurityServiceImpl.hasPermission(667) | Unable to find ACL information for object identity 'gemma.gsec.acl.domain.AclObjectIdentity[Type: ubic.gemma.model.expression.bioAssay.BioAssay; Identifier: 1437]'
[gemma-core 2021-10-12 22:03:50,746] WARN [main] gemma.gsec.SecurityServiceImpl.hasPermission(667) | Unable to find ACL information for object identity 'gemma.gsec.acl.domain.AclObjectIdentity[Type: ubic.gemma.model.expression.bioAssay.BioAssay; Identifier: 1435]'
```

Understanding this issue might help us make the ACL generation more robust.

**Comment by arteymix** [comment]
As of 80d9121df54da3e35a915fe609a08bca9b7c9c8f, these two tests are skipped when the exception is raised.

**Comment by arteymix** [comment]
Interestingly, these two tests start off by removing `GSE96006` from the database if it exists. Some ACLs left-overs might cause subsequent tests to fail.

**Comment by arteymix** [comment]
I'm adding `ubic.gemma.core.loader.expression.DataUpdaterTest.testLoadRNASeqDataWithMissingSamples` to the list of affected tests by this.

I noticed that all the errors are related to `GeoServiceImpl` and the service layer performs multiple database operations without a single transaction. Since this is a likely cause of such error (i.e. left-overs/inconsistent state in the ACLs), I marked its operations as transactional in https://github.com/PavlidisLab/Gemma/commit/91184a7b657ac68e18b46d8704e588e57e52d2f6.
